pyramid_google_login/views.py

```
@@ -112,6 +112,8 @@ def callback(request):
     user_logged_in = UserLoggedIn(request, userid, oauth2_token, userinfo)
     try:
         request.registry.notify(user_logged_in)
+    except AuthFailed as e:
```

I'm not a big fan of expecting Pyramid subscribers to raise exceptions because that breaks proper event processing. Also note that we are actually masking the error message as to not leak internal data to the client side.

The whole redirection thing is already a big hack, so I'm not sure we should continue to build upon that; it's not terribly hard to do proper login checks on the client side.